### PR TITLE
[#150563820] Report CIMB - security question when placing a trade

### PR DESCRIPTION
**Pull request by guillaumedebavelaere**

### CODE REVIEW

File: TradeItIosTicketSDK2/TradeItConnector.swift [hunk 1]
```diff
@@ -74,6 +74,10 @@ internal extension TradeItConnector {
 
         if result?.isError() == true { // Server sent an ERROR response so try create a TradeItErrorResult
             result = TradeItResultTransformer.transform(targetClassType: TradeItErrorResult.self, json: json)
+        } else if result?.isSecurityQuestion() == true {
+            result = TradeItResultTransformer.transform(targetClassType: TradeItSecurityQuestionResult.self, json: json)
+        } else if result?.isReviewOrder() == true {
+            result = TradeItResultTransformer.transform(targetClassType: TradeItPreviewOrderResult.self, json: json)
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItConnector.swift]: Are there any other possible values for this result status?

Comment (guillaumedebavelaere) [reply TradeItIosTicketSDK2/TradeItConnector.swift]: No the other one is success but is not handle here

File: TradeItIosTicketSDK2/TradeItOrder.swift [hunk 2]
```diff
@@ -155,9 +159,37 @@ public typealias TradeItPlaceOrderHandlers = (_ onSuccess: @escaping (TradeItPla
     }
 
     private func generatePlaceOrderCallback(tradeService: TradeItTradeService, previewOrderResult: TradeItPreviewOrderResult) -> TradeItPlaceOrderHandlers {
-        return { onSuccess, onFailure in
+        return { onSuccess, onSecurityQuestion, onFailure in
             let placeOrderRequest = TradeItPlaceTradeRequest(orderId: previewOrderResult.orderId)
-            tradeService.placeTrade(placeOrderRequest, onSuccess: onSuccess, onFailure: onFailure)
+            let placeResponseHandler = YCombinator { handler in
+            { (result: TradeItResult?) in
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItOrder.swift]: Indentation is messed up here

File: TradeItIosTicketSDK2/TradeItTradeService.swift [hunk 3]
```diff
@@ -33,8 +33,7 @@
 
     func placeTrade(
         _ data: TradeItPlaceTradeRequest,
-        onSuccess: @escaping (TradeItPlaceTradeResult) -> Void,
-        onFailure: @escaping (TradeItErrorResult) -> Void
+        withCompletionBlock completionBlock: @escaping (TradeItResult) -> Void
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItTradeService.swift]: Why not `onSuccess`, `onSecurityQuestion`, `onFailure`?

Comment (jsom) [reply TradeItIosTicketSDK2/TradeItTradeService.swift]: Hmm ok let's not worry about it now

**Comment by guillaumedebavelaere** [comment]
@jsom is it good to merge ?

**Pull request merged**